ansible/verify_config.py

```diff
@@ -0,0 +1,466 @@
+        self.assertion.assert_true(lambda: self._check_if_inv_name_has_inv_file(),
+                                   reason="Inventory should have an inventory file")
+
+    def _inv_name_from_devices_files_must_be_the_same_as_graph_group_yml_file(self):
+        inv_name_from_devices_files: Set[str] = Utility.get_inv_name_from_file(Config.FANOUT_DEVICES_FILE.value)
+        inv_name_from_graph_group_yml_file: Set[str] = set(Utility.parse_yml(self.file))
+
+        differences = inv_name_from_devices_files ^ inv_name_from_graph_group_yml_file
+
+        if len(differences) != 0:
+            self.assertion.add_error_details(
+                "These are the group names that are not consistent between the 2 files: "
+                f"{Formatting.red(', '.join(differences))}")
+            return False
+
+        return True
+
+    def _check_if_inv_name_has_inv_file(self):
+        is_valid = True
+        inv_name_from_graph_group_yml_file: Set[str] = set(Utility.parse_yml(Config.FANOUT_GRAPH_GROUP_FILE.value))
+        inv_files = set([f for f in os.listdir('.') if os.path.isfile(f)])
+
+        for inv_name in inv_name_from_graph_group_yml_file:
+            if inv_name not in inv_files:
+                is_valid = False
+                self.assertion.add_error_details(
+                    f"'{Formatting.red(inv_name)}' is declared in "
+                    f"{Formatting.bold(Config.FANOUT_GRAPH_GROUP_FILE.value)}"
+                    f"but does not have inventory file. Consider creating '{Formatting.bold('ansible/' + inv_name)}'")
+        return is_valid
+
+
+class FanoutLinkValidator(Validator):
+    def __init__(self):
+        super().__init__(None)
+
+    def validate(self):
+        params = [
+            {"name": "Link file", "file": Config.FANOUT_LINKS_FILE.value},
+            {"name": "Pdu link file", "file": Config.FANOUT_PDU_LINKS_FILE.value},
+            {"name": "Bmc link file", "file": Config.FANOUT_BMC_LINKS_FILE.value},
+            {"name": "Console link file", "file": Config.FANOUT_CONSOLE_LINKS_FILE.value}
+        ]
+
+        for param in params:
+            self.file = param["file"]
+            self.assertion.assert_true(lambda: self._links_file_should_have_equivalent_devices_file(param),
+                                       reason=f"{param['name']} should have its equivalent devices file")
+            self.assertion.assert_true(lambda: self._devices_in_links_file_should_be_in_devices_file(param),
+                                       reason=f"{param['name']} devices does not exist in its devices file")
+
+    def _links_file_should_have_equivalent_devices_file(self, param: Dict[str, str]):
+        inv_name_from_links_files: Set[str] = Utility.get_inv_name_from_file(param["file"])
+        inv_name_from_devices_files: Set[str] = Utility.get_inv_name_from_file(Config.FANOUT_DEVICES_FILE.value)
+
+        differences = inv_name_from_links_files - inv_name_from_devices_files
```

Comment:
Are we checking if two sets contain the same elements or not? If so, I think we should always use `^` instead of `-`

Reply:
For this one as discussed, I changed from `-` to `^`. However for the `_devices_in_links_file_should_be_in_devices_file` im keeping it as `-` because I think that devices file (e.g: `sonic_lab_devices`) can potentially have more declared devices than it's being used in `sonic_lab_links`). Please correct me if my understanding is correct.